src/MPC_Module.jl

```diff
@@ -343,8 +343,8 @@ function simIPlant!(n)
 
   if isequal(n.s.mpc.mode,:OCP)
    if isequal(n.mpc.v.evalNum,1)
-    U = 0*Matrix{Float64}(n.ocp.control.pts,n.ocp.control.num)
-    t = Vector(linspace(t0,tf,n.ocp.control.pts))
+    U = 0*Matrix{Float64}(undef, n.ocp.control.pts,n.ocp.control.num)
```

Comment:
What is going on here? We assign an uninitialized matrix to U, and then a few lines below we assign the return value to the variable U?

Reply:
we pass U to the model
https://github.com/JuliaMPC/NLOptControl.jl/blob/1f95cfce6bd28c9812302b62b787df8eac5d587a/src/MPC_Module.jl#L364

which then returns another U. Here is an example of a model()
https://github.com/JuliaMPC/NLOptControl.jl/blob/master/test/mpc.jl#L3

The first time it runs, I assume that the control signals are zero, because it has not had a chance to solve the problem yet. This helps when moving to real time systems. That is, in simulation, we can solve the problem and then pass it to the plant, pause the plant time then continue. But on a real system, this is not possible. So, I assume that when you start your real system you don't want to send control signals without having solved the problem. This is used for predicting the next initial state. That is, if you assume that during the first run there will be zeros passed to the plant as you solve the problem, then you can use that to predict where the plant will be after some fixed execution horizon. 

This idea is used in updateX0
https://github.com/JuliaMPC/NLOptControl.jl/blob/master/src/MPC_Module.jl#L432

where you might want to predictX0
https://github.com/JuliaMPC/NLOptControl.jl/blob/master/src/MPC_Module.jl#L401

That is predict where to initialize the next OCP, given that by the time the problem solves, the state will have evolved. 

Does this clear things up?

Reply:
check page 44 for more detail on predicting the initial state if you want
https://deepblue.lib.umich.edu/bitstream/handle/2027.42/149859/febbo_1.pdf?sequence=1&isAllowed=y

Reply:
Ok, I think we should just change this to
```julia
U = fill(0.0, n.ocp.control.pts,n.ocp.control.num)
```
instead of
```julia
U = 0*Matrix{Float64}(undef, n.ocp.control.pts,n.ocp.control.num)
```
then, random (undefined) numbers times 0 might be dangerous here.